src/ngraph/op/tensor_iterator.cpp

```diff
@@ -118,41 +161,184 @@ op::TensorIterator::ConcatOutputDescription::ConcatOutputDescription(uint64_t bo
+        AttributeAdapter(std::vector<std::shared_ptr<op::TensorIterator::InputDescription>>& ref)
+        : m_ref(ref)
+    {
+    }
+
+    bool AttributeAdapter<std::vector<std::shared_ptr<op::TensorIterator::InputDescription>>>::
+        visit_attributes(AttributeVisitor& visitor, const std::string& name)
+    {
+        visitor.start_structure(name);
+        int64_t size = m_ref.size();
+        visitor.on_attribute("size", size);
+        ostringstream index;
+        for (int64_t i = 0; i < size; i++)
+        {
+            index.clear();
+            index << i;
+            visitor.on_attribute(index.str(), m_ref[i]);
+        }
+        visitor.finish_structure();
+        return true;
+    }
+
+    template <>
+    FactoryRegistry<op::v0::TensorIterator::OutputDescription>&
+        FactoryRegistry<op::v0::TensorIterator::OutputDescription>::get()
+    {
+        static FactoryRegistry<op::v0::TensorIterator::OutputDescription> registry;
+        static mutex init_guard;
+        // TODO: Add a lock
+        if (registry.m_factory_map.size() == 0)
+        {
+            lock_guard<mutex> guard(init_guard);
+            if (registry.m_factory_map.size() == 0)
+            {
+                registry.register_factory<op::v0::TensorIterator::ConcatOutputDescription>();
+                registry.register_factory<op::v0::TensorIterator::BodyOutputDescription>();
+            }
+        }
+        return registry;
+    }
+
+    constexpr DiscreteTypeInfo AttributeAdapter<
+        std::vector<std::shared_ptr<op::TensorIterator::OutputDescription>>>::type_info;
+
+    constexpr DiscreteTypeInfo
+        AttributeAdapter<std::shared_ptr<op::TensorIterator::OutputDescription>>::type_info;
+
+    AttributeAdapter<std::vector<std::shared_ptr<op::TensorIterator::OutputDescription>>>::
+        AttributeAdapter(std::vector<std::shared_ptr<op::TensorIterator::OutputDescription>>& ref)
+        : m_ref(ref)
+    {
+    }
+
+    bool AttributeAdapter<std::vector<std::shared_ptr<op::TensorIterator::OutputDescription>>>::
+        visit_attributes(AttributeVisitor& visitor, const std::string& name)
+    {
+        visitor.start_structure(name);
+        int64_t size = m_ref.size();
+        visitor.on_attribute("size", size);
+        ostringstream index;
+        for (int64_t i = 0; i < size; i++)
+        {
+            index.clear();
+            index << i;
+            visitor.on_attribute(index.str(), m_ref[i]);
+        }
+        visitor.finish_structure();
+        return true;
+    }
+}
+
+bool op::v0::TensorIterator::visit_attributes(AttributeVisitor& visitor)
+{
+    // We assume the ops in body lambda are alread registered
+    ParameterVector body_parameters;
+    ResultVector body_results;
+    if (m_body)
+    {
+        body_parameters = m_body->get_parameters();
+        body_results = m_body->get_results();
+    }
+    visitor.on_attribute("body_parameters", body_parameters);
```

Comment:
`TensorIterator` doesn't contain "body_parameters" attribute (https://docs.openvinotoolkit.org/latest/_docs_ops_infrastructure_TensorIterator_1.html). I don't fully understand how this method read the body attribute.

Reply:
How do you get the results/parameters associated with the body? I need to make sure there's a way for you to do that.

Reply:
In order to get results/parameters for body, in IR reader we just parse all body and operations with Result type are body results, the operations with Parameter type is body parameters.